system-tests/lib/crecli/commands.go

```diff
@@ -16,7 +16,97 @@ import (
 type CompilationResult struct {
 	WorkflowURL string
 	ConfigURL   string
-	SecretsURL  string
+}
+
+func CompileWorkflow(creCLICommandPath, workflowFolder string, configFile *string, settingsFile *os.File) (CompilationResult, error) {
+	var outputBuffer bytes.Buffer
+
+	// the CLI expects the workflow code to be located in the same directory as its `go.mod`` file. That's why we assume that the file, which
+	// contains the entrypoint method is always named `main.go`. This is a limitation of the CLI, which we can't change.
+
+	compileArgs := []string{"workflow", "compile", "-S", settingsFile.Name()}
+	if configFile != nil {
+		compileArgs = append(compileArgs, "-c", *configFile)
+	}
+	compileArgs = append(compileArgs, "main.go")
+	compileCmd := exec.Command(creCLICommandPath, compileArgs...) // #nosec G204
+	compileCmd.Stdout = &outputBuffer
+	compileCmd.Stderr = &outputBuffer
+	compileCmd.Dir = workflowFolder
+	err := compileCmd.Start()
+	if err != nil {
+		return CompilationResult{}, errors.Wrap(err, "failed to start compile command")
+	}
+
+	err = compileCmd.Wait()
+	fmt.Println("Compile output:\n", outputBuffer.String())
+	if err != nil {
+		return CompilationResult{}, errors.Wrap(err, "failed to wait for compile command")
+	}
+
+	re := regexp.MustCompile(`Gist URL=([^\s]+)`)
+	matches := re.FindAllStringSubmatch(outputBuffer.String(), -1)
+
+	ansiEscapePattern := `\x1b\[[0-9;]*m`
+	re = regexp.MustCompile(ansiEscapePattern)
+
+	result := CompilationResult{}
+
+	expectedGistURLs := 1
+	if configFile != nil {
```

Comment:
Not obvious. Why?

Reply:
binaryURL is always expected, others are optional (some workflows might require them, others not)